src/main/java/co/org/osso/api/CalculatorService.java

```diff
@@ -0,0 +1,60 @@
+package co.org.osso.api;
+
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CalculatorService {
+
+
+    public List getFibonacci(Integer number) {
+
+        List<Integer> fibonacci = new ArrayList<>();
```

Comment:
Mini-interview questions:
What can you tell me about the theoretical performance here? In particular, does the performance depend upon the size of `number`? If so, how is the CPU affected? And how is memory affected? Is there a way we can improve any of them?

You can just reply to this comment to answer - don't need to make any changes to the code.

Reply:
Well, here I go:

In theorerical performance the accepted thing is that a sum of n numbers takes n steps or O(n) running time. As I'm going over the n-2 numbers, that is practically the same. Now, I know that in pratical terms, in order to sum up large numbers more computations are required for be summing up more bits, but I don't know how to calculate that variances and it seems is not important for the crowd unless that be an application requirement.

So I think that practically larger numbers affect but theoretically not, because theory don't take in account the variations in speed between processors and over the time.

I think this affects to the CPU performance maybe slowling down a little (really little I think) because from what I know the mathematical operations are executed in the CPU/ALU and a bigger number should take more cycles of processing. For other side, this affects RAM in the allocation of resources since bigger numbers takes longer memory positions based on the specified datatype in programming language that can handle those numeric values and after overpass the maximum allowed number for the datatype, should came a memory overflow if language programmin not handle this adequately (I guess).

In order to improve, as now is linear time, this should be considered a Logaritmic time solution or even better a Constant Time one.... thinking about how to implement that...

Reply:
And as right now is linear, as the number of digit grows, longer the runtime will be. So not the best implementation still not the worst (exponential runtime).